### PR TITLE
quick cut of RamanLines plugin

### PR DESCRIPTION
**Pull request by mzieg**
This isn't perfect, but it was adequate for a quick test (I needed it to use it :-)

Bugs:

- de-checked compounds don't disappear from legend
- combobox would be more scalable than checkboxes

### CODE REVIEW

File: pluginExamples/Prod/RamanLines.py
[the remark this answers is not in the document]
Comment (mzieg) [reply pluginExamples/Prod/RamanLines.py:1]: I will try to fix this week, between other things.  It's the only way I'll learn :-)

**Comment by mzieg** [comment]
I converted to immediate-mode based on my understanding. You're right, that's MUCH shorter, simpler and more readable :-)

Bugs:
- phantom legend entries remain after de-selecting compounds
- plugin no longer works after connecting, using, disconnecting, and reconnecting

**Comment by samiebee43** [comment]
https://github.com/WasatchPhotonics/ENLIGHTEN/pull/127/commits/475fdf06029cb7eb84ad085a52d330524ba89ebb is a past commit that I thought addressed the non-clearing (phantom) legend issue. This commit is included in mzieg-ramanlines.

The area of interest is in PluginController.py on lines 1187 and 1188. The idea was to not populate config.series_names for undeclared curves. In an immediate mode plugin we want to regenerate the legend on every-frame, and so storing it anywhere leads to the unwanted persistence.  Immediate mode plugins always use undeclared curves, and OOP plugins from our examples do not used undeclared curves--so this change should not significantly impact backwards compatibility.

In Enlighten.out we can see that "found undeclared curve" only happens twice (for air) when it should happen once per frame. Line 1163 of PluginController.py we have "blank any missing series". Instead of blanking it we should completely remove the key from self.plugin_curves. 

I am not sure if defining and undefining curves is a common practice among OOP plugins. If it's not then this continues to be a somewhat backward compatible change. We should make a plan to drop support for legacy plugins before this becomes a serious issue -- maintaining both paradigms as the arch continues to grow adds lots of complexity

**Comment by samiebee43** [comment]
Fixed phantom legend.  Disconnect/Reconnect is not reproducing on my end anymore. Maybe those issues were related.

I think we are good and ready to merge this

**Comment by mzieg** [comment]
> Fixed phantom legend. Disconnect/Reconnect is not reproducing on my end anymore. Maybe those issues were related.
> 
> I think we are good and ready to merge this

Well, it kind of works.  If you enable cyclohexane lines, they appear, and if you disable cyclohexane, they disappear.  However, you can never enable cyclohexane again.  Basically, each named curve can be displayed and hidden "once" :-/

**Comment by samiebee43** [comment]
Starting with some hints about enable -> disable -> enable.

enable ethanol
```
PluginController    graphing series Ethanol 1055.000cm⁻¹
PluginController    found undeclared curve Ethanol 1055.000cm⁻¹...adding. in_legend=True
Graph                     added raw curve 'Ethanol 1055.000cm⁻¹'
PluginController    updating curve Ethanol 1055.000cm⁻¹
Graph                plotting 2 x values [1055, 1055] .. [1055, 1055]
Graph                plotting 2 y values [2240.0, 7700.6] .. [2240.0, 7700.6]
PluginController     graphing series Ethanol 1090.000cm⁻¹

PluginController     found undeclared curve Ethanol 1090.000cm⁻¹...adding. in_legend=True
Graph                add_curve: added a <class 'pyqtgraph.graphicsItems.PlotDataItem.PlotDataItem'> ...
Graph                added raw curve 'Ethanol 1090.000cm⁻¹'
Configuration        Configuration.set: (baseline_correction, algo, AirPLS)
Stylesheets          applying stylesheet light[gray_gradient_button] to widget pushButton_ramanCorrection
PluginController     taking x_values from dict
PluginController     x_values is 1090..1090
PluginController     updating curve Ethanol 1090.000cm⁻¹
Graph                plotting 2 x values [1090, 1090] .. [1090, 1090]
Graph                plotting 2 y values [2240.0, 7700.6] .. [2240.0, 7700.6]
PluginController     graphing series Ethanol 1280.000cm⁻¹

PluginController     found undeclared curve Ethanol 1280.000cm⁻¹...adding. in_legend=True
# It repeats while Ethanol graph is active
```

Uncheck ethanol
```
PluginController    configured series Ethanol 1055.000cm⁻¹ missing
PluginController    configured series Ethanol 1090.000cm⁻¹ missing
PluginController    configured series Ethanol 1280.000cm⁻¹ missing
PluginController    configured series Ethanol 1460.000cm⁻¹ missing
PluginController    configured series Ethanol 880.000cm⁻¹ missing
```

Check it again
```
PluginController    configured series Ethanol 1055.000cm⁻¹ missing
PluginController    configured series Ethanol 1090.000cm⁻¹ missing
PluginController    configured series Ethanol 1280.000cm⁻¹ missing
PluginController    configured series Ethanol 1460.000cm⁻¹ missing
PluginController    configured series Ethanol 880.000cm⁻¹ missing
```

This shows that self.plugin_curves was maintaining an key with the unchecked curve's name.

**Comment by mzieg** [comment]
Hi Samie, this seems to work great in my testing.  Thanks for the help!

**Pull request merged**